### DISCUSSION TITLE
Model checkpointing.

### DISCUSSION THREAD

**Discussion by kuntiik** [Lightning-AI/pytorch-lightning#12778] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello there,
Has anyone encountered an error, when you save a .ckpt file and when you try to load it (for example using torch.load()) you get: 'Model' object has no attribute 'param_groups_fn'. This happens when torch tries to unpickle the file. 
I've been working with PL for a while and so far had no problem with checkpointing. Can somebody please point me to documentation, on how to solve this or provide any insights into this error? 
Thank you

**Comment by rohitgr7** [comment]
hey @kuntiik 

haven't encountered this error before. Can you reproduce it with [BoringModel](https://colab.research.google.com/github/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report/bug_report_model.ipynb). If it comes up with distributed training, you can still share a minimal script.
